packages/graphql/tests/integration/unwind-create/unwind-create.int.test.ts

```diff
@@ -0,0 +1,572 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Driver } from "neo4j-driver";
+import { int } from "neo4j-driver";
+import { graphql } from "graphql";
+import { generate } from "randomstring";
+import Neo4j from "../neo4j";
+import { Neo4jGraphQL } from "../../../src/classes";
+import { generateUniqueType } from "../../utils/graphql-types";
+
+describe("unwind-create", () => {
```

Comment:
Very likely some other test suites test the same logic as the tests below, although consider the following points:

1. The previous tests were written testing different operations in a different test, this worked before as every operation was isolated in a CALL block, now instead is very likely to see side effects from one operation to another.
2. As we silently roll back to the previous implementation when this optimization is not available, a lot of tests are testing the previous implementation rather than the new one.

Reply:
To be fair, I think these tests need to be improved to test a more diverse batch,
For instance:
Create a list of movies, some with some properties and some with others, some with related actors, some with edge properties, and some with the actor website as well.

Reply:
I don't understand the first point, what do you mean by side effects from one operation to another?

My point was that I'm pretty sure we are already testing some of these operations (definitely creating 2 elements and 2 elements with nested) making some of these redundant. If there is a case not covered in `create.int` by all means, keep that, but most of these are definitely covered

Reply:
What I mean with the first point is this, in the previous implementation an operation like creating a movie setting the field name will results in:
```
CALL {
	CREATE (n:Movie) SET n.name = $param.name
}
```
If you then try to create two movie with a single GraphQL request then:
```
CALL {
	CREATE (n:Movie) SET n.name = $param.name
} // little side-effect between the two operation
CALL {
	CREATE (n:Movie) SET n.anotherField = $param.anotherField
}
```
Now it becomes:
```
UNWIND $movies as movies
CREATE (n:Movie) 
SET n.name = $param.name
SET n.anotherField = $param.anotherField  // it may be a side-effect
```
Now in the example above we know that it's unlikely to be a side-effect, but when you merge very different inputs like autogenerated one, from the `@populatedBy`, etc... I'm not that confident that there is no case of side-effect forward in this development.

For instance, in the `create.int`, you may think that the `pringles` tests will cover this scenario but in reality, it will test only the previous implementation as there is a `connect` operation in the complex input.

Reply:
Given that these are contained within a single file, these can easily be removed at a future date when the implementation is more holistic. Having a few redundant tests isn't a _massive_ issue, and I think at this stage, merging is more important to get some feedback from users and the reporting customer.